### DISCUSSION TITLE
readonly user as default in examples

### DISCUSSION THREAD

**Discussion by nomandera** [lldap/lldap#596] [Ideas]
Currently most of the examples are documented with the admin user in mind.

It is likely that in most instances this highest privileged account is not needed and the `readonly` account would suffice.

Suggest standardising on using `readonly` and only documenting `admin` where required perhaps, citing what would stop working if `readonly` was used regardless.

It is not a matter of if but when someones random app is breached and their admin ldap creds are acquired. We might as well reduce users risk since it likely will have little to no functionality cost.

**Comment by nitnelave** [comment]
Definitely. Some early examples were created before the finer-grained permissions were introduced, but yeah, a lot of them are "whatever works", which is already a step up from nothing.

If you have any of the services installed, I'd encourage you to try their configurations with the different permissions and send PRs with the changes. You can follow this rule of thumb:

  -  create/delete/modify users: lldap_admin
  -  change user passwords: lldap_password_manager
  -  just list users and check their login: lldap_strict_readonly

**Reply by nomandera** [reply]
Just to clarify and make sure my older install has the same users:

Default groups:

    create/delete/modify users: lldap_admin
    change user passwords: lldap_password_manager
    just list users and check their login: lldap_strict_readonly

Default users of these groups:

    create/delete/modify users: admin
    change user passwords: none (no default members of lldap_password_manager)
    just list users and check their login: readonly

**Reply by nitnelave** [reply]
There is no default member of lldap_strict_readonly, you can create the one(s) you want and add them there. You can even create one per integration.

**Comment by nomandera** [comment]
Closing as I think we have come to an agreement and I can see PRs already moving to readonly.

Nice work.
